packages/env/lib/build-docker-compose-config.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' );
+const path = require( 'path' );
+
+/**
+ * @typedef {import('./config').Config} Config
+ */
+
+/**
+ * Creates a docker-compose config object which, when serialized into a
+ * docker-compose.yml file, tells docker-compose how to run the environment.
+ *
+ * @param {Config} config A wp-env config object.
+ * @return {Object} A docker-compose config object, ready to serialize into YAML.
+ */
+module.exports = function buildDockerComposeConfig( config ) {
+	const pluginMounts = config.pluginSources.flatMap( ( source ) => [
+		`${ source.path }:/var/www/html/wp-content/plugins/${ source.basename }`,
+
+		// If this is is the Gutenberg plugin, then mount its E2E test plugins.
+		// TODO: Implement an API that lets Gutenberg mount test plugins without this workaround.
+		...( fs.existsSync( path.resolve( source.path, 'gutenberg.php' ) ) && [
+			`${ source.path }/packages/e2e-tests/plugins:/var/www/html/wp-content/plugins/gutenberg-test-plugins`,
+			`${ source.path }/packages/e2e-tests/mu-plugins:/var/www/html/wp-content/mu-plugins`,
```

Comment:
When I run the script with my own plugin, I get the following error on this line:

```sh
✖ undefined is not a function


 TypeError: undefined is not a function
    at /Users/noah.allen/source/gutenberg/packages/env/lib/build-docker-compose-config.js:27:15
    at Array.flatMap (<anonymous>)
    at buildDockerComposeConfig (/Users/noah.allen/source/gutenberg/packages/env/lib/build-docker-compose-config.js:20:44)
    at initConfig (/Users/noah.allen/source/gutenberg/packages/env/lib/env.js:223:14)
    at async start (/Users/noah.allen/source/gutenberg/packages/env/lib/env.js:36:18) 
```

My config looks like:
```json
{
	"plugins": [ "./full-site-editing-plugin", "../../../gutenberg" ],
	"themes": [ "../../../themes/varia", "../../../themes/maywood" ]
}
```

Everything links correctly if I comment out these lines (lines 25-28), which leads me to think it's not handling different cases.

Reply:
Looks like your node version is missing some core libs.

Reply:
Odd! If you run `node` from the command line and enter `require( 'fs' ).existsSync` and `require( 'path' ).resolve` do you see `undefined`?

Reply:
No, they do not display undefined. :) 

<img width="324" alt="Screen Shot 2020-02-06 at 9 42 54 PM" src="https://user-images.githubusercontent.com/6265975/74004095-a5a8a080-4929-11ea-9c90-d244a7c8cbcc.png">

I'm not sure why anything would be weird with the node libs -- they were installed with `nvm`.

I did some initial debugging when I saw this error (adding console.log) around everything in that part of the script, and all of the following were defined:

- config.pluginSources.flatMap
- fs.existsSync
-  path.resolve

But the script still failed with that error. Definitely odd, since there doesn't appear to be a different function there.

Reply:
Fixed the issue here: https://github.com/WordPress/gutenberg/pull/20113